File: aoc_2019_python/day01.py
import aoc

def main(input_lines):
    part1 = 0
    part2 = 0
    for line in input_lines:
        f = get_required_fuel(int(line))
        part1 += f
        while f > 0:
            f = get_required_fuel(f)
            part2 += max(f, 0)
    part2 += part1
    return part1, part2

def get_required_fuel(weight):
    return weight // 3 - 2

aoc.run_lines(main, "day01.txt")

File: aoc_2018_java/src/Day01.java
import java.util.List;
import java.util.Set;
import java.util.HashSet;

public class Day01 implements AocSolver {
    public static void main(String[] args) {
        Aoc.runLines(new Day01());
    }

    public void solve(List<String> inputLines) {
        int part1 = 0;
        for (String line : inputLines) {
            part1 += Integer.valueOf(line);
        }

        Integer part2 = null;
        int frequency = 0;
        Set<Integer> seen = new HashSet<>();
        while (part2 == null) {
            for (String line : inputLines) {
                frequency += Integer.valueOf(line);
                if (seen.contains(frequency)) {
                    part2 = frequency;
                    break;
                }
                seen.add(frequency);
            }
        }

        System.out.println("Part 1: " + part1);
        System.out.println("Part 2: " + part2);
    }
}